src/viam/__init__.py

```diff
@@ -41,6 +41,22 @@ def _log_exceptions(exctype, value, traceback):
 ##################
 # MONKEY PATCHES #
 ##################
-_ResourceName.__str__ = lambda self: f"{self.namespace}:{self.type}:{self.subtype}/{self.name}"
-_ResourceName.__repr__ = lambda self: f"<viam.proto.common.ResourceName {str(self)} at {hex(id(self))}>"
-_ResourceName.__hash__ = lambda self: hash(str(self))

```

Comment:
(q) out of curiosity, what was wrong with the lambda approach?

Reply:
Nothing was wrong with it, but it was not possible to add types to the lambdas. I actually preferred the lambdas because it's so concise, but alas